rct229/rulesets/ashrae9012019/section19/section19rule4.py

```diff
@@ -0,0 +1,321 @@
+                    design_cooling_multiplier_sch_b
+                    == most_used_weekday_hourly_schedule_b
+                )
+
+                return {"inf_pass_cooling_b": inf_pass_cooling_b}
+
+            class SpaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section19Rule4.RuleSetModelInstanceRule.ZoneRule.SpaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": [
+                                "lighting_space_type",
+                                "ventilation_space_type",
+                                "occupant_multiplier_schedule",
+                                "interior_lighting",
+                                "miscellaneous_equipment",
+                            ],
+                        },
+                    )
+
+                def is_applicable(self, context, data=None):
+                    space_b = context.baseline
+                    lighting_space_type_b = space_b["lighting_space_type"]
+                    ventilation_space_type_b = space_b["ventilation_space_type"]
+
+                    return (
+                        lighting_space_type_b == LIGHTING_SPACE.DWELLING_UNIT
+                        or ventilation_space_type_b
+                        == VENTILATION_SPACE.TRANSIENT_RESIDENTIAL_DWELLING_UNIT
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    inf_pass_cooling_b = data["inf_pass_cooling_b"]
+                    day_of_week_for_january_1 = data["day_of_week_for_january_1"]
+                    schedule_b = data["schedule_b"]
+                    space_b = context.baseline
+
+                    # check occupancy schedule
+                    multiplier_sch_occ_b = schedule_b[
+                        space_b["occupant_multiplier_schedule"]
+                    ]
+                    multiplier_sch_occ_hourly_value_b = multiplier_sch_occ_b[
+                        "hourly_values"
+                    ]
+                    design_cooling_multiplier_sch_b = multiplier_sch_occ_b[
+                        "hourly_cooling_design_day"
+                    ]
+
+                    most_used_weekday_hourly_schedule_b = (
+                        get_most_used_weekday_hourly_schedule(
+                            multiplier_sch_occ_hourly_value_b, day_of_week_for_january_1
+                        )
+                    )
+                    occ_pass_cooling_b = (
+                        most_used_weekday_hourly_schedule_b
+                        == design_cooling_multiplier_sch_b
+                    )
+
+                    # check interior lighting
+                    int_lgt_pass_cooling_b = False
+                    for interior_lighting_b in space_b["interior_lighting"]:
```

Comment:
Replace with `find_all` once removed the `interior_lighting` from the `required_field`

Reply:
Addressed.